tf-psa-crypto/include/psa/crypto.h

```diff
@@ -4812,6 +4812,354 @@ psa_status_t psa_verify_hash_abort(
+ *                              \c psa_key_agreement_iop_t, and be inactive.
+
+ * \param private_key           Identifier of the private key to use. It must
+ *                              allow the usage #PSA_KEY_USAGE_DERIVE.
+ * \param[in] peer_key          Public key of the peer. It must be in the
+ *                              same format that psa_import_key() accepts. The
+ *                              standard formats for public keys are documented
+ *                              in the documentation of psa_export_public_key().
+ *                              The peer key data is parsed with the type
+ *                              #PSA_KEY_TYPE_PUBLIC_KEY_OF_KEY_PAIR(type) where
+ *                              type is the type of \p private_key, and with the
+ *                              same bit-size as \p private_key.
+ * \param peer_key_length       Size of \p peer_key in bytes.
+ *
+ * \param alg                   The key agreement algorithm to compute
+ *                              (a \c PSA_ALG_XXX value such that
+ *                              #PSA_ALG_IS_KEY_AGREEMENT(\p alg) is true).
+ *
+ * \param[in] attributes        The attributes for the new key.
+ *                              This function uses the attributes as follows:
+ *                              * The key type must be one of
+ *                                `PSA_KEY_TYPE_DERIVE`,`PSA_KEY_TYPE_RAW_DATA`,
+ *                                `PSA_KEY_TYPE_HMAC`, or
+ *                                `PSA_KEY_TYPE_PASSWORD`.
+ *                              * Implementations must support the
+ *                                `PSA_KEY_TYPE_DERIVE` and
+ *                               `PSA_KEY_TYPE_RAW_DATA` key types.
+ *                              * The size of the returned key is always the
+ *                                bit-size of the shared secret, rounded up to a
+ *                                whole number of bytes. The key size in \p
+ *                                attributes can be zero; if it is nonzero, it
+ *                                must be equal to the output size of the key
+ *                                agreement, in bits.
+ *                              * The output size, in bits, of the key agreement
+ *                                is #PSA_RAW_KEY_AGREEMENT_OUTPUT_SIZE(type,
+ *                                bits), where type and bits are the type and
+ *                                bit-size of \p private_key.
+ *                              * The key permitted-algorithm policy is required
+ *                                for keys that will be used for a cryptographic
+ *                                operation. The key usage flags define what
+ *                                operations are permitted with the key. The key
+ *                                lifetime and identifier are required for a
+ *                                persistent key.
```

Comment:
The lifetime/identifier sentence is a separate bullet point. But we could omit both the policy part and the persistence part altogether, because it's just how attributes work for all key creation functions. The documentation of `psa_key_attributes_t` is enough.

Reply:
As above, updated to latest spec. I left in the parts about lifetime and policy (although much reduced) as this is not particularly mentioned in the docs for `psa_key_attribtutes_t` - I agree that it should be, but various rules mentioned here are glossed over there.